runbotics/runbotics-orchestrator-ui/src/src-app/components/tables/Table/Table.tsx

```diff
@@ -48,9 +48,31 @@ const Table = <T extends object>({
         setIsLoading(loading);
     }, [loading]);
 
-    const data = useMemo(() => {
-        if (subRowProperty) return propData.map((row) => ({ ...row, subRows: row[subRowProperty] }));
+    const replaceKeyRecursive = (tableData: T | T[], newKey: string) => {
+        if (typeof tableData !== 'object' || tableData === null) {
+            return tableData;
+        }
+      
+        if (Array.isArray(tableData)) {
+            return tableData.map(item => replaceKeyRecursive(item, newKey));
+        }
+      
+        const newObj = {};
+      
+        for (const key in tableData) {
+            if (Object.hasOwnProperty.call(tableData, key)) {
+                const newKeyString = key === subRowProperty ? newKey : key;
+                newObj[newKeyString] = replaceKeyRecursive(tableData[key] as T[], newKey);
+            }
```

Comment:
what does it do? can we make it more readable for future devs?

Reply:
So the recursive function ProcessInstance.utils.ts file creates the tree structure where nested objects are stored as 'subprocesses' arrays. In this recursive function we want to bind those subprocesses to nested rows of table. They are organized in correct structure already, but we need to rename its' keys from value passed as subRowProperty props (in that case -'subprocesses') to 'subRows'.